app/main.py

```diff
@@ -1 +1,39 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self,
+                 name: str,
+                 health: int = 100,
+                 hidden: bool = False) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+
+        if health > 0:
+            Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        animal = {
+            "Name": self.name,
+            "Health": self.health,
+            "Hidden": self.hidden
+        }
+
+        return str(animal).replace("'", "")
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    def bite(self, herbivore: Herbivore) -> None:
+        if herbivore.hidden or isinstance(herbivore, Carnivore):
+            return
+
+        herbivore.health -= 50
+
+        if herbivore.health <= 0:
+            died_index = super().alive.index(herbivore)
+            del super().alive[died_index]
```

Comment:
super().alive is not a valid way to access the alive attribute of the parent class. Instead, you should access it through the class itself (Animal.alive), as class attributes are shared among all instances of the class.